frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,209 @@
+          identifier = {
+            ...identifier,
+            warning: true,
+          };
+          continue;
+        }
+        if (identifier.defaultCount.toString().at(0) === '-') {
+          parentWarningMessage = `Default count for ${
+            identifier.resourceType ?? identifier.displayName
+          } cannot be negative.`;
+          identifier = {
+            ...identifier,
+            warning: true,
+          };
+          continue;
+        }
+        const [minCount] = splitValueUnit(
+          identifier.minCount.toString(),
+          determineUnit(identifier),
+          true,
+        );
+        const [maxCount] = splitValueUnit(
+          identifier.maxCount.toString(),
+          determineUnit(identifier),
+          true,
+        );
+        const [defaultCount] = splitValueUnit(
+          identifier.defaultCount.toString(),
+          determineUnit(identifier),
+          true,
+        );
+        if (minCount > maxCount) {
+          parentWarningMessage = `Minimum allowed ${
+            identifier.resourceType ?? identifier.displayName
+          } cannot exceed maximum allowed ${identifier.resourceType ?? identifier.displayName}.`;
+          identifier = {
+            ...identifier,
+            warning: true,
+          };
+        } else if (defaultCount < minCount || defaultCount > maxCount) {
+          parentWarningMessage = `The default count for ${
+            identifier.resourceType ?? identifier.displayName
+          } must be between the minimum allowed ${
+            identifier.resourceType ?? identifier.displayName
+          } and maximum allowed ${identifier.resourceType ?? identifier.displayName}.`;
+          identifier = {
+            ...identifier,
+            warning: true,
+          };
+        } else {
+          identifier = {
+            ...identifier,
+            warning: false,
+          };
+        }
+      } catch (e) {
+        parentWarningMessage = `The resource count for ${
+          identifier.resourceType ?? identifier.displayName
+        } has an invalid unit.`;
+        identifier = {
+          ...identifier,
+          warning: true,
+        };
+      }
+    }
+
+    return {
+      ...profile,
+      spec: {
+        ...profile.spec,
+        enabled:
+          profile.spec.enabled &&
+          (!parentWarningMessage ||
+            parentWarningMessage === HARDWARE_PROFILES_MISSING_CPU_MEMORY_MESSAGE),
+        warning: parentWarningMessage
+          ? {
+              title: `${!hasCPUandMemory(identifiers) ? 'Incomplete' : 'Invalid'} ${
+                isDefaultProfile === true ? 'default hardware' : 'hardware'
+              } profile`,
+              message: `${parentWarningMessage} ${
+                isDefaultProfile
+                  ? HARDWARE_PROFILES_DEFAULT_WARNING_MESSAGE
+                  : 'Edit the profile to make the profile valid.'
+              }`,
+            }
+          : undefined,
+      },
+    };
+  });
```

Comment:
we dont want to be injecting the warnings into the spec.

instead lets create some new object type like `HardwareProfileValidation` type... better yet we can use zod and create a schema to validate this. 

 thwn we could do things like 

```tsx
export const isHardwareProfileValid = (data: HardwareProfileKind): boolean => {
  const schema = createHardwareProfileValidationSchema(data.selectedProfile);
  const result = schema.safeParse(data);
  return result.success;
};
```